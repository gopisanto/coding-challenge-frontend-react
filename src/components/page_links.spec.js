import React from 'react';
import renderer from 'react-test-renderer';

import PageLinks from './page_links';

describe('<PageLinks />', () => {
  test('renders links to each page', () => {
    const component = renderer.create(<PageLinks totalRecords={35} noOfRecordsPerPage={10} />).toJSON();

    expect(component).toMatchSnapshot();
  });

  test('all navigations are disabled when there is only 1 page', () => {
    const component = renderer.create(<PageLinks totalRecords={5} noOfRecordsPerPage={10} />).toJSON();

    expect(component).toMatchSnapshot();
  })
});
