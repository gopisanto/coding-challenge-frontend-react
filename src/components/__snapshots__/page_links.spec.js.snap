// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLinks /> all navigations are disabled when there is only 1 page 1`] = `
<div
  className="page-links"
>
  <button
    disabled={true}
    onClick={[Function]}
  >
    &lt;&lt; First
  </button>
  <button
    disabled={true}
    onClick={[Function]}
  >
    &lt; Prev
  </button>
  <button
    disabled={true}
    onClick={[Function]}
    value={1}
  >
    1
  </button>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Next &gt;
  </button>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Last &gt;&gt;
  </button>
</div>
`;

exports[`<PageLinks /> renders links to each page 1`] = `
<div
  className="page-links"
>
  <button
    disabled={true}
    onClick={[Function]}
  >
    &lt;&lt; First
  </button>
  <button
    disabled={true}
    onClick={[Function]}
  >
    &lt; Prev
  </button>
  <button
    disabled={true}
    onClick={[Function]}
    value={1}
  >
    1
  </button>
  <button
    disabled={false}
    onClick={[Function]}
    value={2}
  >
    2
  </button>
  <button
    disabled={false}
    onClick={[Function]}
    value={3}
  >
    3
  </button>
  <button
    disabled={false}
    onClick={[Function]}
    value={4}
  >
    4
  </button>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Next &gt;
  </button>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Last &gt;&gt;
  </button>
</div>
`;
